src/main/java/ch/njol/skript/expressions/ExprPotionEffectTier.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Potion Effect Tier")
+@Description("An expression to obtain the amplifier of a potion effect applied to an entity.")
+@Examples("if the amplifier of haste of player >= 3:")
+@Since("INSERT VERSION")
+public class ExprPotionEffectTier extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprPotionEffectTier.class, Number.class, ExpressionType.COMBINED,
+			"[the] [potion] (tier|amplifier|level) of %potioneffecttype% (of|for|on) %livingentity%"
+		);
+	}
+
+	private Expression<PotionEffectType> typeExpr;
+	private Expression<LivingEntity> entityExpr;
+
+	@Override
```

Comment:
This requires a suppress warning as we're casting an object to an assumed generic.
```suggestion
	@SuppressWarnings("unchecked")
	@Override
```

Reply:
That wouldn't effect anything production wise though would it? The PR works intended on my server without it

Reply:
No, but we can suppress the warnings so they don't show up when running code checks and such in an IDE. The casting being done here is safe so :)

Also, the `@SuppressWarnings` should go *after* the `@Override`

Reply:
SuppressWarnings goes before Override in ecj which is what Skript is built on which is also how the majority of syntaxes are formatted. I just checked and there are a few that have it after.

Reply:
There is a clear standard, it's in the code conventions:
![image](https://user-images.githubusercontent.com/29547183/180734877-6927f110-b620-4c06-aa63-439519140320.png)